unstructured/ingest/connector/mongodb.py

```diff
@@ -78,39 +90,209 @@ def to_dict(
         return d
 
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def generate_client(self) -> "MongoClient":
+        from pymongo import MongoClient
+        from pymongo.server_api import ServerApi
+
+        if self.uri:
+            return MongoClient(self.uri, server_api=ServerApi(version=SERVER_API_VERSION))
+        else:
+            return MongoClient(
+                host=self.host,
+                port=self.port,
+                server_api=ServerApi(version=SERVER_API_VERSION),
+            )
+
+    def get_collection(self, client):
+        database = client[self.database]
+        return database.get_collection(name=self.collection)
+
 
 @dataclass
-class MongoDBWriteConfig(WriteConfig):
-    database: str
+class MongoDBDocumentMeta:
     collection: str
+    document_id: str
+    date_created: str
 
 
 @dataclass
-class MongoDBDestinationConnector(BaseDestinationConnector):
-    write_config: MongoDBWriteConfig
-    connector_config: SimpleMongoDBStorageConfig
+class MongoDBIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    connector_config: SimpleMongoDBConfig
+    document_meta: MongoDBDocumentMeta
+    document: dict = field(default_factory=dict)
+    registry_name: str = "mongodb"
+
+    @property
+    def filename(self):
+        return (
+            Path(self.read_config.download_dir)
+            / self.connector_config.collection
+            / f"{self.document_meta.document_id}.txt"
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        return (
+            Path(self.processor_config.output_dir)
+            / self.connector_config.collection
+            / f"{self.document_meta.document_id}.json"
+        )
+
+    def update_source_metadata(self, **kwargs):
+        if self.document is None:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+            return
+        self.source_metadata = SourceMetadata(
+            date_created=self.document_meta.date_created,
+            exists=True,
+        )
+
+    @SourceConnectionError.wrap
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    @BaseSingleIngestDoc.skip_if_file_exists
+    def get_file(self):
+        pass
+
+    @property
+    def record_locator(self) -> t.Optional[t.Dict[str, t.Any]]:
+        return {
+            "host": self.connector_config.host,
+            "collection": self.connector_config.collection,
+            "document_id": self.document_meta.document_id,
+        }
+
+
+@dataclass
+class MongoDBIngestDocBatch(BaseIngestDocBatch):
+    connector_config: SimpleMongoDBConfig
+    ingest_docs: t.List[MongoDBIngestDoc] = field(default_factory=list)
+    list_of_ids: t.List[str] = field(default_factory=list)
+    registry_name: str = "mongodb_batch"
+
+    @property
+    def unique_id(self) -> str:
+        return ",".join(sorted(self.list_of_ids))
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def _get_docs(self):
```

Comment:
Can we add a return type here to make it explicit what sort of data this method is giving back.